source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -440,19 +441,44 @@ bool Filter::parseTlvs(const uint8_t* buf, size_t len) {
     absl::string_view tlv_value(reinterpret_cast<char const*>(buf + idx), tlv_value_length);
     auto key_value_pair = config_->isTlvTypeNeeded(tlv_type);
     if (nullptr != key_value_pair) {
-      ProtobufWkt::Value metadata_value;
-      // Sanitize any non utf8 characters.
-      auto sanitised_tlv_value = MessageUtil::sanitizeUtf8String(tlv_value);
-      metadata_value.set_string_value(sanitised_tlv_value.data(), sanitised_tlv_value.size());
-
-      std::string metadata_key = key_value_pair->metadata_namespace().empty()
-                                     ? "envoy.filters.listener.proxy_protocol"
-                                     : key_value_pair->metadata_namespace();
-
-      ProtobufWkt::Struct metadata(
-          (*cb_->dynamicMetadata().mutable_filter_metadata())[metadata_key]);
-      metadata.mutable_fields()->insert({key_value_pair->key(), metadata_value});
-      cb_->setDynamicMetadata(metadata_key, metadata);
+      if (Runtime::runtimeFeatureEnabled(
+              "envoy.reloadable_features.use_typed_metadata_in_proxy_protocol_listener")) {
+        auto& typed_filter_metadata = (*cb_->dynamicMetadata().mutable_typed_filter_metadata());
```

Comment:
Is typed_filter_metadata also available to the lua filter under dynamicMetadata call? Or does it need to be exposed separately?

Reply:
Lua filter currently only [processes](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/lua/wrappers.cc#L268) untyped metadata from dynamic metadata in steam_info object. To access typed metadata in dynamic metadata the invocation should be:
```
const auto& metadata = streamInfo().dynamicMetadata().typed_filter_metadata();
...
``` 
So, Lua filter code also needs an update.

Reply:
Okay, I can add that for connection stream info

Reply:
Hi @agrawroh, is there a way to access typed filter metadata from Lua today?

Reply:
@kollektiv I have a WIP PR. Will try to get it reviewed and merge